core/policy/service_test.go

```diff
@@ -425,6 +425,116 @@ func (s *ServiceTestSuite) TestPolicyRequirements() {
 			})
 		}
 	})
+
+	s.Run("valid requirements", func() {
+		expectedResource := &domain.Resource{
+			ID:           1,
+			ProviderType: "provider-type-test",
+			ProviderURN:  "provider-urn-test",
+		}
+		expectedProvider := &domain.Provider{}
+		additionalAppeals := []*domain.AdditionalAppeal{
+			{
+				Resource: &domain.ResourceIdentifier{
+					ID: 1,
+				},
+				Role: "viewer",
+			},
+		}
+
+		testCases := []struct {
+			name         string
+			requirements []*domain.Requirement
+		}{
+			{
+				name: "requirement condition on ProviderType only",
+				requirements: []*domain.Requirement{
+					{
+						On: &domain.RequirementTrigger{
+							ProviderType: "test-bigquery",
+						},
+						Appeals: additionalAppeals,
+					},
+				},
+			},
+			{
+				name: "requirement condition on Role only",
+				requirements: []*domain.Requirement{
+					{
+						On: &domain.RequirementTrigger{
+							Role: "test-viewer",
+						},
+						Appeals: additionalAppeals,
+					},
+				},
+			},
+			{
+				name: "appeal identifier using provider type+urn and resource type+urn",
+				requirements: []*domain.Requirement{
+					{
+						On: &domain.RequirementTrigger{
+							Role: "test-viewer",
+						},
+						Appeals: []*domain.AdditionalAppeal{
+							{
+								Resource: &domain.ResourceIdentifier{
+									ProviderType: "test-provider-type",
+									ProviderURN:  "test-provider-urn",
+									Type:         "test-type",
+									URN:          "test-urn",
+								},
+								Role: "viewer",
+							},
+						},
+					},
+				},
+			},
+		}
+
+		for _, tc := range testCases {
```

Comment:
should each test cases has it's own `s.Run`?
CMIIW but right now `tc.name` is not being used anywhere.

Reply:
@StewartJingga updated